mtp_api/apps/security/tests/test_views.py

```diff
@@ -126,16 +148,100 @@ def test_list_credits_for_sender(self):
         )
-        for credit in credits:
+        for credit in sender.credits.all():
             self.assertTrue(credit.id in [d['id'] for d in data])
 
 
-class PrisonerProfileListTestCase(SecurityViewTestCase):
+class RecipientProfileListTestCase(SecurityViewTestCase):
+    def _get_url(self, *args, **kwargs):
+        return reverse('recipientprofile-list')
+
+    def test_filter_by_prisoner_count(self):
+        data = self._get_list(
+            self._get_authorised_user(),
+            prisoner_count__gte=3,
+        )['results']
+        prisoner_counts = Disbursement.objects.filter(
+            method=DISBURSEMENT_METHOD.BANK_TRANSFER,
+            resolution=DISBURSEMENT_RESOLUTION.SENT
+        ).values(
+            'sort_code', 'account_number', 'roll_number',
+        ).order_by(
+            'sort_code', 'account_number', 'roll_number',
+        ).annotate(prisoner_count=Count('prisoner_number', distinct=True))
+
+        prisoner_counts = prisoner_counts.filter(prisoner_count__gte=3)
+
+        self.assertEqual(
+            len(prisoner_counts), len(data)
+        )
+
+    def test_filter_by_prison(self):
+        data = self._get_list(self._get_authorised_user(), prison='IXB')['results']
+
+        recipient_profiles = RecipientProfile.objects.filter(
+            prisoners__prisons__nomis_id='IXB',
+            bank_transfer_details__isnull=False
+        ).distinct()
+
+        self.assertEqual(len(data), recipient_profiles.count())
+        for recipient in recipient_profiles:
+            self.assertTrue(recipient.id in [d['id'] for d in data])
+
+    def test_filter_by_multiple_prisons(self):
+        data = self._get_list(self._get_authorised_user(), prison=['IXB', 'INP'])['results']
+
+        recipient_profiles = RecipientProfile.objects.filter(
+            Q(prisoners__prisons__nomis_id='IXB') |
+            Q(prisoners__prisons__nomis_id='INP'),
+            bank_transfer_details__isnull=False
+        ).distinct()
+
+        self.assertEqual(len(data), recipient_profiles.count())
+        for recipient in recipient_profiles:
+            self.assertTrue(recipient.id in [d['id'] for d in data])
 
+    def test_filter_by_monitoring(self):
+        user = self._get_authorised_user()
+        profiles = RecipientProfile.objects.filter(
+            bank_transfer_details__isnull=False
+        )[:2]
+
+        for profile in profiles:
+            profile.bank_transfer_details.first().recipient_bank_account.monitoring_users.add(user)
+
+        recipient_profiles = RecipientProfile.objects.filter(
+            bank_transfer_details__recipient_bank_account__monitoring_users=user
+        )
+        data = self._get_list(user, monitoring=True)['results']
+
+        self.assertEqual(len(data), recipient_profiles.count())
+        for recipient in recipient_profiles:
+            self.assertTrue(recipient.id in [d['id'] for d in data])
+
+
+class RecipientDisbursementListTestCase(SecurityViewTestCase):
```

Comment:
I would call this `RecipientProfileDisbursementListTestCase` to make is consistent with the view name